idaes/models/unit_models/tests/test_heat_exchanger_1D.py

```diff
@@ -274,36 +275,197 @@ def test_config():
 
 
 @pytest.mark.unit
-def test_config_validation():
+def test_config_validation(caplog):
     m = ConcreteModel()
     m.fs = FlowsheetBlock(dynamic=False)
 
     m.fs.properties = BTXParameterBlock(valid_phase="Liq")
 
     with pytest.raises(ConfigurationError):
-        m.fs.HX_co_current = HX1D(
+        m.fs.HX_counter_current1 = HX1D(
             hot_side={
                 "property_package": m.fs.properties,
+                "transformation_method": "dae.finite_difference",
+            },
+            cold_side={
+                "property_package": m.fs.properties,
+                "transformation_method": "dae.collocation",
+            },
+            flow_type=HeatExchangerFlowPattern.countercurrent,
+        )
+    caplog.clear()
+    with caplog.at_level(idaeslog.INFO):
+        m.fs.HX_countercurrent2 = HX1D(
+            hot_side={
+                "property_package": m.fs.properties,
+            },
+            cold_side={
+                "property_package": m.fs.properties,
+            },
+            flow_type=HeatExchangerFlowPattern.countercurrent,
+        )
+    assert (
+        "Discretization method was "
+        "not specified for the hot side of the "
+        "heat exchanger. "
+        "Defaulting to finite "
+        "difference method on the hot side."
+    ) in caplog.text
+    assert (
+        m.fs.HX_countercurrent2.config.hot_side.transformation_method
+        == "dae.finite_difference"
+    )
+    assert (
+        "Discretization method was "
+        "not specified for the cold side of the "
+        "heat exchanger. "
+        "Defaulting to finite "
+        "difference method on the cold side."
+    ) in caplog.text
+    assert (
+        m.fs.HX_countercurrent2.config.cold_side.transformation_method
+        == "dae.finite_difference"
+    )
+    assert (
+        "For cold_side, a BACKWARD scheme was chosen to discretize the length domain. "
+        "However, this scheme is not an upwind scheme for countercurrent flow, and "
+        "as a result may run into numerical stability issues. To avoid this, "
+        "use a FORWARD scheme (which may result into energy conservation issues "
+        "for coarse discretizations) or use a high-order collocation method."
+    ) in caplog.text
+    caplog.clear()
+
+    with caplog.at_level(idaeslog.INFO):
+        m.fs.HX_countercurrent3 = HX1D(
+            hot_side={
+                "property_package": m.fs.properties,
+                "transformation_method": "dae.finite_difference",
                 "transformation_scheme": "BACKWARD",
             },
             cold_side={
                 "property_package": m.fs.properties,
+                "transformation_method": "dae.finite_difference",
                 "transformation_scheme": "FORWARD",
             },
+            flow_type=HeatExchangerFlowPattern.countercurrent,
+        )
+    assert "To avoid" not in caplog.text
```

Comment:
What is line 352 checking for?

Reply:
This is to check that the upwinding disclaimer is not logged. I'll add a note to that effect.